src/aleph/tcp.clj

```diff
@@ -30,8 +30,14 @@
 
       :exception-caught
       ([_ ctx ex]
-        (when-not (instance? IOException ex)
-          (log/warn ex "error in TCP server")))
+       (cond
+         ;; do not need to log an entire stack trace when SSL handshake failed
+         (netty/ssl-handshake-error? ex)
+         (log/warn "SSL handshake failure:"
+                   (.getMessage ^Throwable (.getCause ^Throwable ex)))
+
+         (not (instance? IOException ex))
+         (log/warn ex "error in TCP server")))
```

Comment:
Any reason the channel is closed everywhere but here?

Reply:
[The commit I picked this from](https://github.com/clj-commons/aleph/commit/7aedd987acc2ff4d914c74243132be86c40c68c9) only touched the client handlers. Looks like the `error-handling-1.0.0` branch also has [a commit which does a similar thing for server handlers](https://github.com/clj-commons/aleph/commit/54954b6e8fc05e840154d3bc14ffd5d0bf688ae4). We could look into porting this, too, but I'm starting to feel this PR's scope starts creeping a bit too much. All I wanted was less noisy SSL handshake errors and what I got was the whole jungle :smile: How about I drop the `netty/close` commit again and we make a separate PR for salvaging the good parts from `error-handling-1.0.0`?

Reply:
> How about I drop the netty/close commit again and we make a separate PR for salvaging the good parts from error-handling-1.0.0?

Ok, let's handle the `netty/close` on another PR.

Reply:
Just force-pushed an update to drop the commit :+1: